fishy/helper/luaparser.py

```diff
@@ -0,0 +1,69 @@
+from math import floor
+
+
+def _sv_parser(path):
+    try:
+        with open(path, "r") as f:
+            lua = f.read()
+    except FileNotFoundError:
+        return None
+
+    """
+    bring lua saved-var file into a useable format:
+    remove all redundant symbols, make lowercase, split into list of lines
+    """
+    for delimiter in (",","[","]",'"'," "):
+        lua = lua.replace(delimiter,"")
+    lua = lua.replace("=\n{","={").lower().split("\n")
+
+    """
+    the lua saved-var file is parsed to a tree of dicts
+    each line represents either one node in the tree or the end of a subtree
+    the last symbol of each line decides the type of the node (branch vertex or leaf)
+    """
+    stack = []
+    root = (dict(),"root")
+    stack.append(root)
+    for line in lua:
+        if line == "":
+            break
+        if line[-1] == '{': #subtree start
+            t = dict()
+            tname = line.split("=")[0]
+            stack.append((t,tname))
+        elif line[-1] == '}': #subtree end
+            t = stack.pop()
+            tp = stack.pop()
+            tp[0][t[1]] = t[0]
+            stack.append(tp)
+        else: #new element in tree
+            name,val = line.split("=")
+            t = stack.pop()
+            t[0][name] = val
+            stack.append(t)
```

Comment:
y not just use regex?

Reply:
readability

Reply:
![grafik](https://user-images.githubusercontent.com/1882648/116539061-a0e25a80-a8e8-11eb-8971-a49781848401.png)

Reply:
sure then xD... regex is easier, thats y i asked

Reply:
![grafik](https://user-images.githubusercontent.com/1882648/121805189-63dce800-cc4a-11eb-9e7a-eb2dc9657709.png)